nixos/modules/services/web-apps/mealie.nix

```diff
@@ -0,0 +1,145 @@
+{ config, lib, pkgs, ...}:
+let
+  cfg = config.services.mealie;
+  pkg = pkgs.mealie;
+in
+{
+  options.services.mealie = {
+    enable = lib.mkEnableOption "Mealie, a recipe manager and meal planner";
+
+    host = lib.mkOption {
+      type = lib.types.str;
+      default = "0.0.0.0";
+      description = "Host on which the server must listen";
+    };
+
+    url_host = lib.mkOption {
+      type = lib.types.str;
+      default = "localhost";
+      description = "Host on which creating the URL on the service";
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 9000;
+      description = "Port on which to serve the Mealie service";
+    };
+
+    protocol = lib.mkOption {
+      type = lib.types.str;
+      default = "http";
+      description = "Protocol to use to serve the service";
+    };
+
+    logLevel = lib.mkOption {
+      # Taken from https://docs.python.org/3/library/logging.html#logging-levels
+      type = lib.types.enum [ "NOTSET" "DEBUG" "INFO" "WARNING" "ERROR" "CRITICAL" ];
+      default = "INFO";
+      description = "Log level to use for this service";
+    };
+
+    extraConfig = lib.mkOption {
+      type = lib.types.attrs;
+      default = {};
+      description = lib.mdDoc ''
+        Some extra configuration for the service.
+        See [the mealie documentation](https://nightly.mealie.io/documentation/getting-started/installation/backend-config/) for available options and default values.
+      '';
+      example = {
+        ALLOW_SIGNUP = "false";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    services.mealie.extraConfig = {
+      DEFAULT_GROUP = lib.mkDefault "Home";
+      DEFAULT_EMAIL = lib.mkDefault "changeme@example.com";
+      DEFAULT_PASSWORD = lib.mkDefault "MyPassword";
+      ALLOW_SIGNUP = lib.mkDefault "true";
+
+      API_DOCS = lib.mkDefault "False";
+
+      TOKEN_TIME = lib.mkDefault 24;
```

Comment:
I also think those aren't really necessary, as they have default values in the app. We should really only define the ones that _need_ to be defined by the module, the rest should be taken care of by `extraConfig` if the user wants to override them.